pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgres_types.go

```diff
@@ -4,6 +4,70 @@
 
 package v1beta1
 
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type PostgresConfig struct {
+	// Files to mount under "/etc/postgres".
+	// ---
+	// +optional
+	Files []corev1.VolumeProjection `json:"files,omitempty"`
+
+	// Configuration parameters for the PostgreSQL server. Some values will
+	// be reloaded without validation and some cause PostgreSQL to restart.
+	// Some values cannot be changed at all.
+	// More info: https://www.postgresql.org/docs/current/runtime-config.html
+	// ---
+	//
+	// Postgres 17 has something like 350+ built-in parameters, but typically
+	// an administrator will change only a handful of these.
+	// +kubebuilder:validation:MaxProperties=50
+	//
+	// # File Locations
+	// - https://www.postgresql.org/docs/current/runtime-config-file-locations.html
+	//
+	// +kubebuilder:validation:XValidation:rule=`!has(self.config_file) && !has(self.data_directory)`,message=`cannot change PGDATA path: config_file, data_directory`
+	// +kubebuilder:validation:XValidation:rule=`!has(self.external_pid_file)`,message=`cannot change external_pid_file`
+	// +kubebuilder:validation:XValidation:rule=`!has(self.hba_file) && !has(self.ident_file)`,message=`cannot change authentication path: hba_file, ident_file`
+	//
+	// # Connections
+	// - https://www.postgresql.org/docs/current/runtime-config-connection.html
+	//
+	// +kubebuilder:validation:XValidation:rule=`!has(self.listen_addresses)`,message=`network connectivity is always enabled: listen_addresses`
+	// +kubebuilder:validation:XValidation:rule=`!has(self.port)`,message=`change port using .spec.port instead`
+	// +kubebuilder:validation:XValidation:rule=`!has(self.ssl) && !self.exists(k, k.startsWith("ssl_"))`
```

Comment:
For some of these, are you considering the message to be self evident?

Reply:
I don't know what to say in these cases. I welcome any suggestions!